python/pyspark/sql/dataframe.py

```diff
@@ -2200,10 +2200,13 @@ def _collectAsArrow(self):
         .. note:: Experimental.
         """
         with SCCallSiteSync(self._sc) as css:
-            sock_info = self._jdf.collectAsArrowToPython()
+            port, auth_secret, jserver_obj = self._jdf.collectAsArrowToPython()
```

Comment:
Do you think if it makes sense to make a `_serialize_from_jvm` (like `_serialize_to_jvm`)? This could be done separately.

Reply:
Possibly, `_load_from_socket` is basically deserializing from the JVM